From 636e5d7ea31b9e54da626bef3cd045bed19b6938 Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@nxp.com>
Date: Fri, 15 Dec 2017 00:09:25 +0800
Subject: [PATCH 3090/5242] MLK-17128-1 dt-bindings: typec: add max_snk_mw
 property

commit  d0a5aaa8434acdc6b1772b5dc8f7d295939d2513 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add max_snk_mw property for power sink to know the max power.
Because the max mw may be smaller than the max ma multiplied by
max mv:
       max-mw <= max-mv*max-ma

After the power sink decides the PDO from source, it needs
to check the power to see if the provided power of this PDO
can match its requirement, which needs consider max_snk_mw to
know the max current it can have based on selected voltage.

Acked-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Li Jun <jun.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 Documentation/devicetree/bindings/usb/typec.txt |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Documentation/devicetree/bindings/usb/typec.txt b/Documentation/devicetree/bindings/usb/typec.txt
index e71b426..f1339dc 100644
--- a/Documentation/devicetree/bindings/usb/typec.txt
+++ b/Documentation/devicetree/bindings/usb/typec.txt
@@ -28,6 +28,10 @@ Required properties:
                 power is less then it, Capability Mismatch is set, required
                 only for power sink and power dual role with power delivery
                 support.
+- max-snk-mw:   The max power the sink can support in milliwatts, required
+                for power sink and power dual role with power delivery support,
+                power sink needs this property to get the max current based on
+                the selected PDO.
 
 Example:
 
-- 
1.7.9.5

